### DISCUSSION TITLE
Consoling useWatch Value after reset form is still giving old value

### DISCUSSION THREAD

**Discussion by prayasjain1996** [react-hook-form/react-hook-form#9498] [General] [answered]
I have to do some stuff after reset a form. And using useWatch Value after reset Form but it still consoling the old value   

`import React, { useEffect } from "react";
import "./styles.css";
import { useForm, useWatch } from "react-hook-form";

export default function App() {
  const form = useForm({
    defaultValues: {
      test: "",
      name: ""
    }
  });
  const nameField = useWatch({
    name: "test",
    control: form.control,
    defaultValue: ""
  });
  const onFormReset = () => {
    form.reset();
    console.log(nameField, "dh"); // useWatch Value
  };
  return (
    <div className="App">
      <p>Hello CodeSandbox</p>
      <input ref={form.register} name="test" />
      <button type="button" onClick={() => onFormReset()}>
        Do Reset
      </button>
    </div>
  );
}`
Is something wrong I am doing ?
codesandbox - https://codesandbox.io/s/exciting-surf-595j79?file=/src/App.js:0-689 

@bluebill1049 Please help

**Comment by bluebill1049** [comment, marked as answer]
useWatch return value is utilised for render function, not within the callback function, you may want to use `getValues()` instead.
